A27-xds-global-load-balancing.md

```diff
@@ -0,0 +1,432 @@
+`RouteConfiguration` name specified in the `Listener`.
+
+Because we are requesting one specific resource by name, the RDS response
+should include exactly one RouteConfiguration.  However, the gRPC client
+must be tolerant of servers that may ignore the resource name in the
+request; if the server returns multiple resources, the client will look
+for the one with the name it asked for, and it will ignore the rest of
+the entries.
+
+#### RouteConfiguration Proto
+
+The `RouteConfiguration` includes a list of zero or more __VirtualHost__
+resources.  The gRPC client will look through this list to find an element
+whose domains field matches the server name specified in the "xds:" URI
+(with port, if any, stripped off).  If no matching VirtualHost is found in
+the RouteConfiguration, the xds resolver will return an error to the client
+channel.
+
+In our initial implementation, the only field in the VirtualHost proto
+that the gRPC client needs to look at is the list of routes.  The client
+will look only at the last route in the list (the default route),
+whose match field must contain a prefix field whose value is the empty
+string and whose route field must be set.  Inside that route message,
+the cluster field will indicate which cluster to send the request to.
+
+If the cluster cannot be determined, the xds resolver will return an
+error to the client channel.
+
+If the cluster can be determined, the xds resolver will return a service
+config that selects the CDS LB policy.  The LB policy's configuration will
+include the name of the cluster.
+
+Notes on backward compatibility:
+
+- Initially, the gRPC client will look only at the last route in the list,
+  which is expected to be the default route.  In the future, we will add
+  support for multiple routes, which may be selected based on which RPC
+  method is being called or possibly on a header match (details TBD).
+- Initially, the gRPC client will require that the route action specify a
+  single cluster, as described above.  However, in the future, we will add
+  support for the `weighted_clusters` field, which will allow us to support
+  traffic splitting.  (This will require introducing a new top-level LB
+  policy.)
+- The `weighted_clusters` field is in the same "oneof" as the `cluster` field.
+  This means that older clients requiring the `cluster` field will
+  stop working if the `weighted_clusters` field is specified instead.
+  Unfortunately, there does not seem to be a way to avoid this, so it's
+  something that our early users will need to be aware of: when we add
+  support for `weighted_clusters`, they must upgrade their clients before
+  they can start using this feature.  (We might be able to ameliorate
+  this by adding support for route matchers at the same time as we
+  add `weighted_clusters` support, in which case the final route in the
+  list can continue to use the `cluster` field but earlier routes can
+  use `weighted_clusters`.)
```

Comment:
One option might be to add some additional metadata matching, e.g. allowing the route to depend on information in the `Node` expressing capability. Maybe.